samoa-api/src/main/java/org/apache/samoa/streams/kafka/KafkaStream.java

```diff
@@ -0,0 +1,201 @@
+ * #L%
+ */
+
+import com.github.javacliparser.IntOption;
+import com.github.javacliparser.StringOption;
+import org.apache.samoa.instances.*;
+import org.apache.samoa.moa.core.Example;
+import org.apache.samoa.moa.core.FastVector;
+import org.apache.samoa.moa.core.InstanceExample;
+import org.apache.samoa.moa.core.ObjectRepository;
+import org.apache.samoa.moa.options.AbstractOptionHandler;
+import org.apache.samoa.moa.streams.InstanceStream;
+import org.apache.samoa.moa.tasks.TaskMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+public class KafkaStream extends AbstractOptionHandler implements
+        InstanceStream {
+
+    private static final long serialVersionUID = 1L;
+
+    protected InstancesHeader streamHeader;
+
+    protected Instances instances;
+
+    private KafkaReader reader;
+
+    protected InstanceExample lastInstanceRead;
+
+    protected Queue<String> instanceQueue;
+
+    private static final Logger logger = LoggerFactory
+            .getLogger(KafkaStream.class);
+
+    public void KafkaReader() {
+        reader = new KafkaReader();
+    }
+
+    public IntOption classIndexOption = new IntOption("classIndex", 'c',
+            "Class index of data. 0 for none or -1 for last attribute in file.",
+            -1, -1, Integer.MAX_VALUE);
+
+    public IntOption numAttrOption = new IntOption("numNumerics", 'u',
+            "The number of numeric attributes in" +
+                    " dataset", 300, 0,
+            Integer.MAX_VALUE);
+
+    public StringOption topicOption = new StringOption("topic", 't',
+            "Topic in the kafka to be used for reading data", "test");
+
+    public IntOption numMaxreadOption = new IntOption("numMaxread", 'r',
+            "Number of instances to be read in single read from kafka", 1, 0,
+            Integer.MAX_VALUE);
+
+    public IntOption partitionOption = new IntOption("partition", 'n',
+            "Partition number to be used for reading data", 0);
+
+    public IntOption portOption = new IntOption("port", 'p',
+            "Port in kafka to read data", 9092);
+
+    public StringOption seedOption = new StringOption("seed", 's',
+            "Seeds for kafka", "localhost");
+
+    public IntOption numClassesOption = new IntOption("numClasses", 'k',
+            "The number of classes in the data.", 2, 2, Integer.MAX_VALUE);
+
+    public IntOption timeDelayOption = new IntOption("timeDelay", 'y',
+            "Time delay in milliseconds between two read from kafka", 0, 0, Integer.MAX_VALUE);
+
+    @Override
+    protected void prepareForUseImpl(TaskMonitor monitor,
+                                     ObjectRepository repository) {
+        this.reader = new KafkaReader();
+        generateHeader();
+        instanceQueue = new LinkedList<String>();
+        this.lastInstanceRead = null;
+    }
+
+    protected void generateHeader() {
```

Comment:
I am not sure about the design of this class, especially of this method.
Would you care to explain the rationale behind it?